carousel/carousel.js

```diff
@@ -0,0 +1,228 @@
+const carouselAutoplayIntervalMap = new Map();
+
+/*
+    CarouselOptions may include:
+        - allowCycle = Boolean. Default is true. Whether the carousel can loop over items or not.
+        - showDots = Boolean. Default is true. Whether the dots are visible.
+        - buttonStyle = String. Additional style to give each button.
+        - carouselStyle = String. Additional style to give the carousel as a whole.
+        - autoplayTimeInMs = Number. Default is 6000. The time for each carousel to autoplay in ms.
+ */
+function createCarousel(carouselName, carouselItems, carouselOptions) {
+    const carousel = document.createElement("div");
+    carousel.classList.add("carousel");
+
+    if (!carouselOptions) {
+        carouselOptions = {}; // To not have to check for undefined all the time.
+    }
+
+    if ("carouselStyle" in carouselOptions &&
+        typeof carouselOptions.carouselStyle === "string") {
+        carousel.style = carouselOptions.carouselStyle;
+    }
+
+    const carouselTitle = document.createElement("h1");
+    carouselTitle.classList.add("carousel-title");
+    carouselTitle.innerText = carouselName ?? "Carousel Title Missing";
+
+    if ("allowCycle" in carouselOptions &&
+        typeof carouselOptions.allowCycle === "boolean" &&
+        carouselOptions.allowCycle === false) {
+        carousel.dataset.allowCycle = "false";
+    }
+    else {
+        carousel.dataset.allowCycle = "true";
+    }
+
+    const carouselContentHolder = generateCarouselContentHolder(carouselItems);
+    const carouselLeftButton = document.createElement("button");
```

Comment:
Again, when in a context of a function, no need to add prefixes to every variable